pkg/networking/networking.go

```diff
@@ -52,11 +53,50 @@ type defaultHeadersRoundTripper struct {
 	networkAccess            *networkImpl
 }
 
+func (rt *defaultHeadersRoundTripper) logRoundTrip(request *http.Request, response *http.Response, err error) {
+	if rt.networkAccess == nil && rt.networkAccess.logger == nil {
+		return
+	}
+
+	logHeader := http.Header{}
+	loglevel := zerolog.TraceLevel
+
+	if rt.networkAccess.logger.GetLevel() == loglevel {
+		for i, v := range request.Header {
+			for _, value := range v {
+				if strings.ToLower(i) == "authorization" || strings.ToLower(i) == "session-token" {
+					authHeader := strings.Split(value, " ")
+					if len(authHeader) == 2 && len(authHeader[1]) > 4 {
+						value = authHeader[0] + " " + authHeader[1][0:4] + "***"
+					} else {
+						value = "***"
+					}
+				}
+				logHeader.Add(i, value)
+			}
+		}
```

Comment:
`nitpick`: I wonder if it could be possible to put away the loop logic to a function and assign the result to `logHeader`, something like:

```go
logHeader := sanitiseLogHearder(request.Header)
```